app/src/org/commcare/activities/CommCareHomeActivity.java

```diff
@@ -1098,6 +1098,10 @@ private boolean checkAndStartUnsentFormsTask(final boolean syncAfterwards,
     @Override
     protected void onResume() {
         super.onResume();
+        if (redirectedToLogin) {
+            finish();
+            return;
```

Comment:
@phillipm So the point of setting up the whole session aware infrastructure was to avoid causing a `SessionUnvailableException` if the user returns to code that assumes there is an active session, after having been logged out from the app manager. Calling the `redirectToLogin` function doesn't really resolve this problem if the code path is still allowed to continue running, because it might execute a statement that relies on an active session. In fact now that I think about it, this was probably contributing to the confusing crashes we were encountering after uninstalling an app from the manager and then coming back to CommCare. We were both thinking that calling `redirectToLogin` alone was sufficient, but I don't think it is.